Tests/MongoSwiftBenchmarks/BenchmarkUtils.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+import XCTest
+
+extension XCTestCase {
+
+    /// Measure the time for a single execution of operation
+    func measureTime(_ operation: () throws -> Void) throws -> Double {
+        let startTime = ProcessInfo.processInfo.systemUptime
```

Comment:
Is this monotonic and/or high resolution time? It looks like [from this post](https://stackoverflow.com/questions/24755558/measure-elapsed-time-in-swift) we could also be using `DispatchTime.now()`. Are there alternatives?

Reply:
I think that _might_ be OSX/iOS specific though, GCD probably isn't provided by the swift implementation for e.g. linux

Reply:
that post is interesting, it looks like there's also straight up:
```
let start = Date()
try operation()
return start.timeIntervalSinceNow
```

Reply:
I didn't investigate this much and just chose what `XCTest` uses for measuring time ([here](https://github.com/apple/swift-corelibs-xctest/blob/01e6126d1cd1c5a81e7a1695487c09d57261577c/Sources/XCTest/Private/WallClockTimeMetric.swift#L58)). this returns a [`TimeInterval`](https://developer.apple.com/documentation/foundation/timeinterval) (which is really just a `Double`) and has "sub-millisecond precision". looks like `timeIntervalSinceNow` also returns a `TimeInterval`. though I am looking at the [source](https://github.com/apple/swift-corelibs-foundation/blob/04b903a6091562a72846f21b048a966532a37f94/Foundation/Date.swift#L93) for `timeIntervalSinceNow` and it appears to use `CFAbsoluteTimeGetCurrent()`, which is stated to not necessarily be monotonic on repeated calls in its [docs](https://developer.apple.com/documentation/corefoundation/1543542-cfabsolutetimegetcurrent). (but this is a single call for each operation so maybe that's ok?) 

yeah, the `DispatchTime` thing is odd, the [documentation](https://developer.apple.com/documentation/dispatch/dispatchtime) talks about behavior on Apple platforms but its unclear what happens if you try to use it on Linux... though I see a [Swift ticket](https://bugs.swift.org/browse/SR-5706) about some fixes to it, so it looks like it would be available? 

what do you think?

Reply:
👍